src/main/java/rs/raf/demo/controllers/KnjizenjeController.java

```diff
@@ -1,21 +1,112 @@
+    public ResponseEntity<?> createDnevnikKnjizenja(@Valid @RequestBody Knjizenje dnevnikKnjizenja){
+        return ResponseEntity.ok(knjizenjaService.save(dnevnikKnjizenja));
     }
+
+    @PutMapping(consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> updateDnevnikKnjizenja(@Valid @RequestBody Knjizenje dnevnikKnjizenja){
+        Optional<Knjizenje> optionalDnevnik = knjizenjaService.findById(dnevnikKnjizenja.getKnjizenjeId());
+        if(optionalDnevnik.isPresent()) {
+            return ResponseEntity.ok(knjizenjaService.save(dnevnikKnjizenja));
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @DeleteMapping(value = "/{id}")
+    public ResponseEntity<?> deleteDnevnikKnjizenja(@PathVariable("id") Long id){
+        Optional<Knjizenje> optionalDnevnik = knjizenjaService.findById(id);
+        if(optionalDnevnik.isPresent()) {
+            knjizenjaService.deleteById(id);
+            return ResponseEntity.ok().build();
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public ResponseEntity<?> getDnevnikKnjizenjaId(@PathVariable("id") Long id){
+        Optional<Knjizenje> optionalDnevnik = knjizenjaService.findById(id);
+        if(optionalDnevnik.isPresent()) {
+            return ResponseEntity.ok(knjizenjaService.findById(id));
+        } else {
+            return ResponseEntity.notFound().build();
+        }
+    }
+
+    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<?> search(@RequestParam(name = "search") String search,
+                                    @RequestParam(defaultValue = ApiUtil.DEFAULT_PAGE) @Min(ApiUtil.MIN_PAGE) Integer page,
+                                    @RequestParam(defaultValue = ApiUtil.DEFAULT_SIZE) @Min(ApiUtil.MIN_SIZE) @Max(ApiUtil.MAX_SIZE) Integer size,
+                                    @RequestParam(defaultValue = "-datumKnjizenja")  String[] sort){
+        RacunSpecificationsBuilder<Knjizenje> builder = new RacunSpecificationsBuilder<>();
+        Pageable pageSort = ApiUtil.resolveSortingAndPagination(page, size, sort);
+
+        Pattern pattern = Pattern.compile("(\\w+?)(:|<|>)(\\w+?),");
+        Matcher matcher = pattern.matcher(search + ",");
+        while (matcher.find()) {
+            builder.with(matcher.group(1), matcher.group(2), matcher.group(3));
+        }
+
+        Specification<Knjizenje> spec = builder.build();
+
+        try{
+            Page<Knjizenje> result = knjizenjaService.findAll(spec, pageSort);
+
+            if(result.isEmpty()) {
+                return ResponseEntity.notFound().build();
+            }
+
+            return ResponseEntity.ok(result);
+        }
+        catch (RuntimeException e){
```

Comment:
Moze da se izbaci try-catch (baci pogled na MyExceptionHandler klasu, tamo vodimo racuna o exceptionima). To treba da bude standardan format kada bacamo greske (imamo i ErrorResponse klasu)

Trenutno (ERROR 500):
<img width="357" alt="image" src="https://user-images.githubusercontent.com/48336447/160486963-a5a4c0e5-2fb6-40a5-9694-f3fc49b7fd19.png">

Kad obrises try-catch (400 BAD REQUEST):
<img width="411" alt="image" src="https://user-images.githubusercontent.com/48336447/160486866-ebc2d5ff-f84f-4ac8-b789-427cebcee325.png">